src/printer.ts

```diff
@@ -1295,12 +1295,33 @@ export class PugPrinter {
 
 		if (this.previousToken?.type === 'dot') {
 			const lastTagToken: TagToken | undefined = previousTagToken(this.tokens, this.currentIndex);
+			let typeAttributeToken: AttributeToken | undefined;
+			if (lastTagToken != null) {
+				const tokens: Token[] = this.tokens.slice(this.tokens.indexOf(lastTagToken) + 1, this.currentIndex - 1);
+				typeAttributeToken = tokens.find((token) => token.type === 'attribute' && token.name === 'type') as
+					| AttributeToken
+					| undefined;
+			}
 
 			let parser: BuiltInParserName | undefined;
 			switch (lastTagToken?.val) {
-				case 'script':
-					parser = 'babel';
+				case 'script': {
+					if (typeAttributeToken == null) {
+						parser = 'babel';
+						break;
+					}
+					const knownJSTypeValues: Record<string, BuiltInParserName> = {
+						// js
+						"'text/javascript'": 'babel',
+						"'application/javascript'": 'babel',
+						"'application/ecmascript'": 'babel',
+						// json
+						"'application/json'": 'json',
+						"'application/ld+json'": 'json'
+					};
+					parser = knownJSTypeValues[typeAttributeToken.val as string];
```

Comment:
Maybe I will move this to utils :thinking:  
That way I can also only create the _mimetype registry_ once

Reply:
I would also strip the quotes in case someone wishes to use double quotes.

Reply:
You are absolutely right :eyes: 
Haven't thought about this :sweat_smile: